### ISSUE TITLE
Request: be able to select text with Shift + Arrow Keys

### ISSUE THREAD

**Issue by KeronCyst** [Genymobile/scrcpy#942]
Bluetooth keyboards can do this when connected to the device.

**Comment by rom1v** [comment]
I implemented forwarding SHIFT, and injecting COPY on Ctrl+c (if Android >= 7, like I do for paste since [v1.14](https://github.com/Genymobile/scrcpy/releases/tag/v1.14)).

Here is a branch [`copy`](https://github.com/Genymobile/scrcpy/commits/copy).

Unfortunately, there is a bug on Android (at least on my ROM on One Plus 7 Pro).

To reproduce:
 - select any text and click on "copy"
 - click somewhere in the middle of a text in a textfield, it should show a dialog containing `Paste` and `Select all`
 - when this dialog is displayed, press `Shift`+`←` (left)

![dialog](https://user-images.githubusercontent.com/543275/83128983-74bcc900-a0dc-11ea-8762-84860aece9df.png)


On my device, it crash 100% reproducible:
```
E MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
E MessageQueue-JNI: java.lang.IllegalArgumentException
E MessageQueue-JNI: 	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:35)
E MessageQueue-JNI: 	at android.widget.SelectionActionModeHelper$TextClassificationHelper.init(SelectionActionModeHelper.java:1062)
E MessageQueue-JNI: 	at android.widget.SelectionActionModeHelper.resetTextClassificationHelper(SelectionActionModeHelper.java:468)
E MessageQueue-JNI: 	at android.widget.SelectionActionModeHelper.resetTextClassificationHelper(SelectionActionModeHelper.java:476)
E MessageQueue-JNI: 	at android.widget.SelectionActionModeHelper.startSelectionActionModeAsync(SelectionActionModeHelper.java:123)
E MessageQueue-JNI: 	at android.widget.Editor.startSelectionActionModeAsync(Editor.java:2154)
E MessageQueue-JNI: 	at android.widget.Editor.refreshTextActionMode(Editor.java:2099)
E MessageQueue-JNI: 	at android.widget.TextView.spanChange(TextView.java:10719)
E MessageQueue-JNI: 	at android.widget.TextView$ChangeWatcher.onSpanChanged(TextView.java:13446)
E MessageQueue-JNI: 	at android.text.SpannableStringBuilder.sendSpanChanged(SpannableStringBuilder.java:1322)
E MessageQueue-JNI: 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:767)
E MessageQueue-JNI: 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:691)
E MessageQueue-JNI: 	at android.text.Selection.extendSelection(Selection.java:175)
E MessageQueue-JNI: 	at android.text.Selection.extendSelection(Selection.java:167)
E MessageQueue-JNI: 	at android.text.Selection.extendLeft(Selection.java:396)
E MessageQueue-JNI: 	at android.text.method.ArrowKeyMovementMethod.left(ArrowKeyMovementMethod.java:72)
E MessageQueue-JNI: 	at android.text.method.BaseMovementMethod.handleMovementKey(BaseMovementMethod.java:165)
E MessageQueue-JNI: 	at android.text.method.ArrowKeyMovementMethod.handleMovementKey(ArrowKeyMovementMethod.java:65)
E MessageQueue-JNI: 	at android.text.method.BaseMovementMethod.onKeyDown(BaseMovementMethod.java:42)
E MessageQueue-JNI: 	at android.widget.TextView.doKeyDown(TextView.java:8498)
E MessageQueue-JNI: 	at android.widget.TextView.onKeyDown(TextView.java:8229)
E MessageQueue-JNI: 	at android.view.KeyEvent.dispatch(KeyEvent.java:2801)
E MessageQueue-JNI: 	at android.view.View.dispatchKeyEvent(View.java:13435)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1937)
E MessageQueue-JNI: 	at com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:460)
E MessageQueue-JNI: 	at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1874)
E MessageQueue-JNI: 	at android.app.Activity.dispatchKeyEvent(Activity.java:3971)
E MessageQueue-JNI: 	at android.support.v4.app.SupportActivity.superDispatchKeyEvent(ComponentActivity.java:108)
E MessageQueue-JNI: 	at android.support.v4.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
E MessageQueue-JNI: 	at android.support.v4.app.SupportActivity.dispatchKeyEvent(ComponentActivity.java:126)
E MessageQueue-JNI: 	at android.support.v7.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:535)
E MessageQueue-JNI: 	at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
E MessageQueue-JNI: 	at android.support.v7.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:2533)
E MessageQueue-JNI: 	at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:374)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5771)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5639)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5121)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5174)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5140)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5280)
E MessageQueue-JNI: 	at android.view.ViewRootImpl$Input
```

**Comment by rom1v** [comment]
> reported here: https://issuetracker.google.com/issues/157583090

This is fixed in Android 11 :tada:

**Comment by rom1v** [comment]
Fixed by #1598.
